.gitlab-ci-check-golang-unittests-v2.yml

```diff
@@ -0,0 +1,113 @@
+# .gitlab-ci-check-golang-unittests.yml
+#
+# This gitlab-ci template runs unit tests and publishes code coverage
+# from a Go repository
+#
+# Add it to the project in hand through Gitlab's include functionality
+#
+# include:
+#   - project: 'Northern.tech/Mender/mendertesting'
+#     file: '.gitlab-ci-check-golang-unittests.yml'
+#
+# If the repo has some compile time dependencies, the template expect to
+# exist a deb-requirements.txt file with the Debian OS required packages
+#
+# The template uses latest Mender supported golang version.
+# To override the version to use, append to your .gitlab-ci.yml:
+# test:unit:
+#   image: golang:1.11.4
+#
+# Requires the following variables set in the project CI/CD settings:
+#   COVERALLS_TOKEN: Token from coveralls.io for this repository
+#
+# By default, it installs MongoDB version 4.4 for Debian buster. You can
+# specify the Debian release name for the upstream APT repository and
+# the MongoDB version overriding the variables below:
+#
+# test:unit:
+#   variables:
+#     MONGODB_DEBIAN_RELEASE: "stretch"
+#     MONGODB_VERSION: "3.6"
+#
+
+stages:
+  - test
+  - publish
+
+test:unit:
+  tags:
+    - mender-qa-worker-generic-light
+  stage: test
+  needs: []
+  except:
+    - /^saas-[a-zA-Z0-9.]+$/
+  image: golang:1.20-alpine3-17
+  services:
+    - mongo:4.4
+  variables:
+    GIT_STRATEGY: clone # clone entire repo instead of reusing workspace
+    GIT_DEPTH: 0 # avoid shallow clone, this test requires full git history
+    TEST_MONGO_URL: "mongodb://mongo"
+  before_script:
+    # Install JUnit test reporting formatter
+    - go install github.com/jstemmer/go-junit-report@v1.0.0
+
+  script:
+    - >-
+      go test ./... -v -covermode=atomic -coverprofile=coverage.txt 2>&1 |
+      tee /dev/stderr |
+      go-junit-report > test-results.xml || exit $?
+
+  artifacts:
+    expire_in: 2w
+    paths:
+      - coverage.txt
+    reports:
+      junit: test-results.xml
+    when: always
+
+publish:unittests:
+  tags:
+    - mender-qa-worker-generic-light
+  stage: publish
+  except:
+    - /^saas-[a-zA-Z0-9.]+$/
+  image: golang:1.20-alpine3.17
+  dependencies:
+    - test:unit
+  before_script:
+    # Install dependencies
+    - apk add --no-cache git
+    - go install github.com/mattn/goveralls@latest
+    # Coveralls env variables:
+    #  According to https://docs.coveralls.io/supported-ci-services
+    #  we should set CI_NAME, CI_BUILD_NUMBER, etc. But according
+    #  to goveralls source code (https://github.com/mattn/goveralls)
+    #  many of these are not supported. Set CI_BRANCH,
+    #  and pass few others as command line arguments.
+    #  See also https://docs.coveralls.io/api-reference
+    - export CI_BRANCH=${CI_COMMIT_BRANCH}
+  script:
+    - tar -xvf unit-coverage.tar
```

Comment:
This you don't need, the only artifact now is the coverage.txt

Reply:
Oops, forgot to adjust this.